tests/licensedcode/data/plugin_license/text/scan.expected.json

```diff
@@ -21,69 +183,16 @@
-              "licenses": [
-                {
-                  "key": "gpl-2.0",
-                  "name": "GNU General Public License 2.0",
-                  "short_name": "GPL 2.0",
-                  "category": "Copyleft",
-                  "is_exception": false,
-                  "is_unknown": false,
-                  "owner": "Free Software Foundation (FSF)",
-                  "homepage_url": "http://www.gnu.org/licenses/gpl-2.0.html",
-                  "text_url": "http://www.gnu.org/licenses/gpl-2.0.txt",
-                  "reference_url": "https://scancode-licensedb.aboutcode.org/gpl-2.0",
-                  "scancode_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/gpl-2.0.LICENSE",
-                  "spdx_license_key": "GPL-2.0-only",
-                  "spdx_url": "https://spdx.org/licenses/GPL-2.0-only"
-                },
-                {
-                  "key": "linux-syscall-exception-gpl",
-                  "name": "Linux Syscall Exception to GPL",
-                  "short_name": "Linux Syscall Exception to GPL",
-                  "category": "Copyleft Limited",
-                  "is_exception": true,
-                  "is_unknown": false,
-                  "owner": "Linux Foundation",
-                  "homepage_url": null,
-                  "text_url": "",
-                  "reference_url": "https://scancode-licensedb.aboutcode.org/linux-syscall-exception-gpl",
-                  "scancode_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/linux-syscall-exception-gpl.LICENSE",
-                  "spdx_license_key": "Linux-syscall-note",
-                  "spdx_url": "https://spdx.org/licenses/Linux-syscall-note"
-                },
-                {
-                  "key": "linux-openib",
-                  "name": "Linux-OpenIB",
-                  "short_name": "Linux-OpenIB",
-                  "category": "Permissive",
-                  "is_exception": false,
-                  "is_unknown": false,
-                  "owner": "Linux Foundation",
-                  "homepage_url": "https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/infiniband/core/sa.h",
-                  "text_url": "https://github.com/ofiwg/libfabric/blob/master/contrib/buildrpm/README",
-                  "reference_url": "https://scancode-licensedb.aboutcode.org/linux-openib",
-                  "scancode_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/linux-openib.LICENSE",
-                  "spdx_license_key": "Linux-OpenIB",
-                  "spdx_url": "https://spdx.org/licenses/Linux-OpenIB"
-                }
-              ]
+              "rule_url": null
             }
           ]
         }
       ],
       "license_clues": [],
       "percentage_of_license_text": 100.0,
+      "for_licenses": [
```

Comment:
what is a licenses in this context? is this a license detection? if so we have to be explicit ... and also I wonder if we could come with an id that's a little more friendly than a UUID? may be an expression cleaned with the `commoncode.text.python_safe_name` function and padding a dash and the UUID? And if this is about license detections, then IMHO the top level attribute should be also `license_detections` and not a plain licenses

Reply:
Implementing this.